lib/compat/wordpress-6.4/fonts/font-face/class-wp-font-face-resolver.php

```diff
@@ -56,30 +56,54 @@ private static function parse_settings( array $settings ) {
 		foreach ( $settings['typography']['fontFamilies'] as $font_families ) {
 			foreach ( $font_families as $definition ) {
 
-				// Skip if font-family "name" is not defined.
-				if ( empty( $definition['name'] ) ) {
+				// Skip if "fontFace" is not defined, meaning there are no variations.
+				if ( empty( $definition['fontFace'] ) ) {
 					continue;
 				}
 
-				// Skip if "fontFace" is not defined, meaning there are no variations.
-				if ( empty( $definition['fontFace'] ) ) {
+				// Skip if "fontFamily" is not defined.
+				if ( empty( $definition['fontFamily'] ) ) {
 					continue;
 				}
 
-				$font_family = $definition['name'];
+				$font_family_name = static::maybe_parse_name_from_comma_separated_list( $definition['fontFamily'] );
+
+				// Skip if no font family is defined.
+				if ( empty( $font_family_name ) ) {
+					continue;
+				}
 
 				// Prepare the fonts array structure for this font-family.
-				if ( ! array_key_exists( $font_family, $fonts ) ) {
-					$fonts[ $font_family ] = array();
+				if ( ! array_key_exists( $font_family_name, $fonts ) ) {
+					$fonts[ $font_family_name ] = array();
 				}
 
-				$fonts[ $font_family ] = static::convert_font_face_properties( $definition['fontFace'], $font_family );
+				$fonts[ $font_family_name ] = static::convert_font_face_properties( $definition['fontFace'], $font_family_name );
 			}
 		}
 
 		return $fonts;
 	}
 
+	/**
+	 * Parse font-family name from comma-separated lists.
+	 *
+	 * If the given `fontFamily` is a comma-separated lists (example: "Inter, sans-serif" ),
+	 * parse and return the fist font from the list.
+	 *
+	 * @since 6.4.0
+	 *
+	 * @param string $font_family Font family `fontFamily' to parse.
+	 * @return string Font-family name.
+	 */
+	private static function maybe_parse_name_from_comma_separated_list( $font_family ) {
+		if ( str_contains( $font_family, ',' ) ) {
+			$font_family = explode( ',', $font_family )[0];
+		}
+
+		return $font_family;
```

Comment:
```suggestion
		return trim( $font_family, "\"'" );
```
Should this be stripped of wrapping quotes before returning? Some of the testing today included various use cases where this might be a factor. This would also address the unit test failure happening [here](https://github.com/WordPress/gutenberg/actions/runs/6240465587/job/16940632328?pr=54615#step:14:130). (For reference, [`phpunit/data/themedir1/fonts-block-theme/theme.json`](https://github.com/WordPress/gutenberg/blob/74e1db8c535115eecaf8cfc00db8eef3a3d8b9cd/phpunit/data/themedir1/fonts-block-theme/theme.json) uses escaped quotes in both of the `fontFamily` values in question, which with this PR are now the values returned in the test.)